CMakeLists.txt

```diff
@@ -1,174 +1,16 @@
-  set (CMAKE_Fortran_FLAGS_RELEASE "-fast -O3")
-  set (CMAKE_Fortran_FLAGS_DEBUG   "-O0")
-elseif (Fortran_COMPILER_NAME MATCHES "NVHPC")
-  message(STATUS "Setting NVHPC fortran flags")
-  if (ENABLE_OPENACC) 
-    if (OPENACC_TARGET MATCHES "gpu")
-      #set(CMAKE_Fortran_FLAGS "-cpp -Mfree -Kieee -Minfo=accel -g -stdpar=gpu -acc -target=gpu -Minstrument")
-      set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -cpp -Mfree -Kieee -Minfo=accel -stdpar=gpu -gpu=cc${CUDA_ARCH_FIRST},managed -acc -target=gpu")
-      #add_definitions("-lnvhpcwrapnvtx")
-      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -lnvhpcwrapnvtx")
-    else()
-      set(CMAKE_Fortran_FLAGS "-cpp -Mfree -Kieee -Minfo=accel -g -stdpar -acc -target=multicore")
-    endif(OPENACC_TARGET MATCHES "gpu")
-  else()
-    set(CMAKE_Fortran_FLAGS "-cpp -Mfree -Kieee -Minfo=accel -g")
-  endif()
-  set (CMAKE_Fortran_FLAGS_RELEASE "-fast -O3")
-  set (CMAKE_Fortran_FLAGS_DEBUG   "-O0 -g -DDEBG")
-elseif (Fortran_COMPILER_NAME MATCHES "Fujitsu")
-  message(STATUS "Setting Fujitsu fortran flags")
-  set (CMAKE_Fortran_FLAGS "-Cpp")
-  set (CMAKE_Fortran_FLAGS_RELEASE "-O3")
-  set (CMAKE_Fortran_FLAGS_DEBUG "-O0")
-else (Fortran_COMPILER_NAME MATCHES "GNU")
-  message ("CMAKE_Fortran_COMPILER full path: " ${CMAKE_Fortran_COMPILER})
-  message ("Fortran compiler: " ${Fortran_COMPILER_NAME})
-  message ("No optimized Fortran compiler flags are known, we just try -O2...")
-  set (CMAKE_Fortran_FLAGS_RELEASE "-O2")
-  set (CMAKE_Fortran_FLAGS_DEBUG   "-O0 -g")
-endif (Fortran_COMPILER_NAME MATCHES "GNU")
-
-if (CMAKE_BUILD_TYPE MATCHES "DEBUG")
-  add_definitions("-DDEBG")
-endif (CMAKE_BUILD_TYPE MATCHES "DEBUG")
-
-
 find_package(MPI REQUIRED)
-# Stop if there is no MPI_Fortran_Compiler
 if (MPI_Fortran_COMPILER)
-    message(STATUS "MPI_Fortran_COMPILER found: ${MPI_Fortran_COMPILER}")
+  message(STATUS "MPI_Fortran_COMPILER found: ${MPI_Fortran_COMPILER}")
 else (MPI_Fortran_COMPILER)
-    message(SEND_ERROR "This application cannot compile without MPI")
+  message(SEND_ERROR "This application cannot compile without MPI")
 endif(MPI_Fortran_COMPILER)
-# Warning if Include are not found => can be fixed with more recent cmake version
-if (MPI_FOUND)
-    message(STATUS "MPI FOUND: ${MPI_FOUND}")
-    include_directories(SYSTEM ${MPI_INCLUDE_PATH})
-    message(STATUS "MPI INCL ALSO FOUND: ${MPI_INCLUDE_PATH}")
-else (MPI_FOUND)
-    message(STATUS "NO MPI include have been found. The executable won't be targeted with MPI include")
-    message(STATUS "Code will compile but performaces can be compromised")
-    message(STATUS "Using a CMake vers > 3.10 should solve the problem")
-    message(STATUS "Alternatively use ccmake to manually set the include if available")
-endif (MPI_FOUND)
 
+# Add defualt definitions such as Version and Double precision
 execute_process(
   COMMAND git describe --tag --long --always
```

Comment:
Is it possible to have the variable `GIT_VERSION` as described below ?

- "git hash" if the source tree is clean
- "git hash-modif" if the source tree is not clean

Reply:
That was the best command i found to have a VERSION, however is it necessary? I kept since it was there but It is not really doing anything this FLAG. Shall we remove it?

Reply:
In my opinion, it is a good practice to print in the listing of the simulation the version of the code used. The variable GIT_VERSION is used for that purpose. However, currently, the variable does not allow to distinguish cases with a clean source tree and cases with a modified source tree. That would be a nice improvement.

Any chance we can adapt [this](https://github.com/xcompact3d/x3div/blob/Fix_HC_TGV/configure.ac#L208) ?